src/UglyToad.PdfPig.Tokens/StreamToken.cs

```diff
@@ -29,7 +29,34 @@ public StreamToken(DictionaryToken streamDictionary, IReadOnlyList<byte> data)
             StreamDictionary = streamDictionary ?? throw new ArgumentNullException(nameof(streamDictionary));
             Data = data ?? throw new ArgumentNullException(nameof(data));
         }
-        
+
+        /// <inheritdoc />
+        public bool Equals(IToken obj)
+        {
+            if (this == obj)
+                return true;
+
+            if (!(obj is StreamToken other))
+            {
+                return false;
+            }
+
+            if (!StreamDictionary.Equals(other.StreamDictionary))
+            {
+                return false;
+            }
+
+            // Note: This maybe slow?
+            // Maybe pre calculate some sort of hash and compare that?
+            for (var index = 0; index < Data.Count; ++index)
```

Comment:
Like the comment say, should calculate some sort of hash? To avoid full bytes comparison?

Reply:
I think this is fine, since the byte array is already in memory it's not going to be worth hashing unless it happens on a tight loop somewhere, I'll profile it to confirm but I don't think it currently is.